### DISCUSSION TITLE
Authentification proxy support

### DISCUSSION THREAD

**Discussion by KptCheeseWhiz** [autobrr/autobrr#1664] [Ideas]
**Is your feature request related to a problem? Please describe.**
All my microservices are already behind Authelia, an authentification proxy, that already provides a way to authenticate users. Since there is no way to disable authentification in autobrr, I sometimes have to login twice, once through Authelia and another one in autobrr. Yes, this is not really an issue if I have to click on one more button if I haven't been on the page after one month, but it would still be a nice feature to have.

**Describe the solution you'd like**
I'd like the application to use the authentification proxy's headers to authentificate the user or at least, have an option to disable authentification.

**Describe alternatives you've considered**
There are no alternatives

**Additional context**
I implemented the authentification proxy on a [fork](https://github.com/KptCheeseWhiz/autobrr), but I am not sure if it fits the code requirements here (my first time coding in Golang). It works on my setup, up to you guys to see if you want this feature implemented on the main repo.

**Comment by zze0s** [comment]
Hi

It's been discussed before and my stance on allowing auth proxies like Authelia has been that it needs to be more than just a "disable password" option, because people will see that and use it without thinking much, which then leads to unsecured and exposed instances of autobrr which isn't great.

This however, looks like it does a bit more than that, with some extra settings. Well done! And if it's your first time in Golang I'm highly impressed 😄

I need to check it out thoroughly and test it myself. I see one TODO about validating a header etc.

Some concerns etc:

* Will this only work with Authelia, or will it work with eg Authentik as well?
* How well does it work behind a reverse proxy? More headers etc needed?

So in conclusion - not opposed but it needs testing! 👍

**Comment by davidnewhall** [comment]
In notifiarr, the operator switches from the "built in user" to auth proxy support by setting the username to `webauth`. The operator also has to set the `upstreams` list variable so it contains the CIDR or IP of the trusted upstream [auth] proxy. This makes sure the app only accepts the `x-webauth-user` header from designated IPs or networks. Then, for a user to be logged in, the upstream proxy has to pass the header `x-webauth-user: <username>`. This configuration requires the operator to know a little bit about their network, and properly configure the header in their proxy. It supports any auth proxy that can pass the username in a header (which is pretty much all of them).

I believe all the most relevant code can be found in this file: https://github.com/Notifiarr/notifiarr/blob/main/pkg/client/handlers_gui.go

Good luck!

**Comment by TMD20** [comment]
I was hoping that all the arr clients moving to force auth would support oauth.

**Comment by s0up4200** [comment]
We are working on implementing OpenID Connect support and we are looking for testers: https://github.com/autobrr/autobrr/pull/1853

**Comment by pentago** [comment]
Great to see this. If you can provide some basic Authelia instructions, I'lll test drive it for a while and report back.

**Reply by s0up4200** [reply]
Hi. This was merged a while back. Confirmed to work with Authelia without doing anything custom. Should be enough to follow Authelia docs and https://autobrr.com/configuration/authentication#openid-connect-oidc
